src/databasetasks.h

```diff
@@ -47,6 +47,7 @@ class DatabaseTasks : public ThreadHolder<DatabaseTasks>
 		void threadMain();
 	private:
 		void runTask(const DatabaseTask& task);
+		void flushInternal();
```

Comment:
With the tasks queue, I believe, but yes, a free void function.

Also, you are double locking the taskLock now, no?